llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -11882,22 +11880,23 @@ bool ScalarEvolution::isImpliedCondBalancedTypes(
     // Forms 1. and 2. require swapping the operands of one condition. Don't
     // do this if it would break canonical constant/addrec ordering.
     if (!isa<SCEVConstant>(RHS) && !isa<SCEVAddRecExpr>(LHS))
-      return isImpliedCondOperands(FoundPred, RHS, LHS, FoundLHS, FoundRHS,
-                                   CtxI);
+      return isImpliedCondOperands(ICmpInst::getSwappedCmpPredicate(*P), RHS,
```

Comment:
IMO the double `getSwappedCmpPredicate` on `FoundPred` is less clear than just `FoundPred` here.

Reply:
Likewise below

Reply:
Possibly, but what can we do about it? getMatching only returns one CmpPredicate.

Reply:
Can we not just use `FoundPred` instead of `ICmpInst::getSwappedCmpPredicate(*P)`, either way its a nit so do as you prefer.

Reply:
Wouldn't that result in a correctness issue?